crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -85,16 +83,47 @@ pub fn parse_scarb_config(
             "Failed to read Scarb.toml manifest file, not found in current nor parent directories",
         )?;
 
-    let package = &metadata.packages[0].manifest_metadata.tool;
+    let package_tool_sncast_result = get_package_tool_sncast(&metadata);
+    if package_tool_sncast_result.is_err() {
+        return Ok(CastConfig::default());
+    }
+
+    cast_config_from_package_tool_sncast(package_tool_sncast_result.unwrap(), profile)
+}
+
+pub fn get_package_tool_sncast(metadata: &scarb_metadata::Metadata) -> Result<&Value> {
+    let first_package = metadata
+        .packages
+        .get(0)
+        .ok_or_else(|| anyhow!("No package found in metadata"))?;
+
+    let tool = first_package
+        .manifest_metadata
+        .tool
+        .as_ref()
+        .ok_or_else(|| anyhow!("No field [tool] found in package"))?;
+
+    let tool_sncast = tool
+        .get("sncast")
+        .ok_or_else(|| anyhow!("No field [tool.sncast] found in package"))?;
```

Comment:
could probably be merged into one?

Reply:
Definitely could. Just thought it'd look cleaner.

Reply:
I see. Wonder what is the memory and speed footprint on this (genuinely don't know).
Let's search for rust good practices, and try to align to them.

Reply:
I don't think there would be any significance in the impact of this approach on the memory / execution speed.
Also, I didn't find any resources about it :/
I think it comes down to the coding style preferences / readability